supafunc/_async/functions_client.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any, Dict, Literal, Optional, Union
+
+from httpx import HTTPError, Response
+
+from ..errors import FunctionsHttpError, FunctionsRelayError
+from ..utils import AsyncClient, __version__
+
+
+class AsyncFunctionsClient:
+    def __init__(self, url: str, headers: Dict):
+        self.url = url
+        self.headers = {
+            "User-Agent": f"supabase-py/functions-py v{__version__}",
+            **headers,
+        }
+
+    async def _request(
+        self,
+        method: Literal["GET", "OPTIONS", "HEAD", "POST", "PUT", "PATCH", "DELETE"],
+        url: str,
+        headers: Union[Dict[str, str], None] = None,
+        json: Optional[dict[Any, Any]] = None,
+    ) -> Response:
+        async with AsyncClient(base_url=self.url, headers=self.headers) as client:
+            response = await client.request(method, url, json=json, headers=headers)
+            try:
+                response.raise_for_status()
+            except HTTPError as exc:
+                raise FunctionsHttpError(
+                    response.json().get("error")
+                    or f"An error occurred while requesting your edge function at {exc.request.url!r}."
+                )
```

Comment:
```suggestion
                ) from exc
```
`raise ... from` should give more correct error reporting here 
https://stackoverflow.com/questions/24752395/python-raise-from-usage